week4.Rmd

```diff
@@ -0,0 +1,236 @@
+- Apply the basics of DNA ligation/restriction enzymes
+
+# Prerequisites
+- Basics of how plasmids are assembled with restriction enzyme digests
+- Basics of bacteria and DNA
+- Basic python syntax
+
+# Background
+Every day, researchers assemble and edit DNA for an uncountable variety of purposes. In the field of Synthetic Biology, researchers often manipulate the DNA of a microorganism in order to change its phenotype, enabling the organism to conduct behavior it otherwise would not have been able to. The canonical example of this is the creation of a circular DNA assembly, known as a *plasmid*, that contains the human gene for insulin. This plasmid is then inserted into a culture of bacteria, often **E. coli**, and produced on an industrial scale.
+
+But how does one create a plasmid? A plasmid must be able to be "read" by its host bacteria and transcribed into RNA, then the target protein. The challenge here isn't in **synthesizing** the DNA (this can be done in an industrial lab at a relatively low cost), but rather in designing a plasmid logically.
+
+A functional plasmid has a few important components:
+1. A vector backbone that allows insertion/proliferation of the plasmid while also providing antibiotic resistance (for genetic screening)
+2. The gene of interest (GOI)
+
+
+```{python, include = FALSE}
+goi = "ATGAACAATGCACTATGGTCAAAAATGCTGGTGGAAGAGGATCCCGGGTTTTTTAGACATAGATACACTCATCAGCAACAATGCACTATGGTCAAAAATGCTGGTGGAAGAGGATCCCGGGTTTTTTGAGAAACTG"
+backbone = "aggctcagtcgaaagactgggcctttcgttttatctgttgtttgtcggtgaacgctctctactagagtcacactggctcaccttcgggtgggcctttctgcgtttataccttaaatattggtttaaaagcaggttaaaagacaggttagcggtggccgaaaaacgggcggaaacccttgcaaa"
+```
+
+In a hidden code chunk I've loaded in a GOI and vector backbone as `goi`,`backbone`, respectively. You can use a `return` call of these variables to print their values.
+
+NOTE: For ease of viewing, these gene sequences have been significantly abbreviated.
+
+If you've printed `backbone`, you may notice that some of the gene sequence is written in upper-text and some in lower-case. Let's write a function to convert all our genes to upper-case.
+
+```{python}
+def to_upper(seq):
+  # First make sure that the argument is a string
+  if isinstance(seq, str) == False:
+    print("Error: sequence is not a string")
+  # Luckily python has a built-in string operation to convert to upper case, so we don't need to do any work :)
+  else:
+    return seq.upper()
+```
+
+```{python}
+to_upper(1) # Returns an error
+to_upper("abc")
+```
+
+Now we can convert all our strings to upper case:
+
+```{python}
+goi = to_upper(goi)
+backbone = to_upper(backbone)
+# Let's inspect prom
+goi # Looks good!
+```
+
+But wait, doesn't DNA have two strands? Yep! DNA is canonically written 5' to 3', the direction that DNA polymerase travels. We can generate the **complementary strand** of DNA using a simple function. This function will turn each nucleotide into its complement:
+A --> T
```

Comment:
This is genius! What a cool way to use a function!

Reply:
Thanks!